src/main/java/org/folio/service/finance/transaction/PendingToPendingEncumbranceStrategy.java

```diff
@@ -83,4 +92,35 @@ private List<EncumbranceRelationsHolder> getTransactionsToUpdate(List<Encumbranc
     return toUpdate;
   }
 
+  public Future<EncumbrancesProcessingHolder> addPendingPaymentsToUpdate(EncumbrancesProcessingHolder holder,
+      RequestContext requestContext) {
+    if (holder.getEncumbrancesForDelete().isEmpty()) {
+      return Future.succeededFuture(holder);
+    }
+    List<String> idsOfEncumbrancesToDelete = holder.getEncumbrancesForDelete().stream()
+      .map(EncumbranceRelationsHolder::getOldEncumbrance)
```

Comment:
What do you think about adding filter to avoid NPE? Are you sure that oldEncumbrance in any case will be not null?

Reply:
I think we should check that in `getTransactionsToDelete()`. I am not sure why it was not breaking before...

Reply:
Actually I think I know why it was working before: there is always a new encumbrance created in `EncumbranceRelationsHoldersBuilder` for existing fund distributions (even though it won't be used in the pending to pending strategy). So the two (old and new) encumbrances cannot be both null. That change was not strictly needed, but it doesn't hurt to add a little check in case the implementation for `EncumbranceRelationsHoldersBuilder` changes in the future.